src/pymodaq/daq_viewer/daq_viewer_main.py

```diff
@@ -2122,17 +2288,19 @@ def single(self, Naverage=1, args_as_dict={}):
 
     def grab_data(self, Naverage=1, live=True, **kwargs):
         """
-            | Update status with 'Start Grabing' Update_status sub command of the Thread command.
-            | Process events and grab naverage is needed.
 
-            =============== =========== ==================
-            **Parameters**    **Type**    **Description**
-            *Naverage*        int
-            =============== =========== ==================
+        This method is called (indirectly) by DAQ_Viewer.grab_data.
+        Call the detector_plugin.grab_data method.
+
+        !!!Not to be confused with DAQ_Viewer.grab_data!!!
+
+        Parameters
+        ----------
+        Naverage : int
+            ??
+        live : bool
+            ??
```

Comment:
corresponds to a grab (with opposition to a snap)

Reply:
I tried to improve the DAQ_Detector docstrings here: 5809744b1a7b07727435769b031bdd579baaca7f

but I think the software averaging should be tested because the behavior is quite strange. I did the following:
TEST of DAQ_Viewer with the 1D Mock_spectro:

* if I change Naverage nothing happens
* if then I click Show averaging nothing happens
* if I stop/restart I can see an effect of Show averaging but it is quite strange. I would expect to see the average value of my signal with 10 time less refreshing of the viewer. But instead we see with the same refreshing time the signal that is more and more average at each iteration, up to go back to the non-averaged signal, and it will loop like this.
* the "live averaging" button seems nice.

It is not the purpose of this PR so I let this for further work.